### PR TITLE
Release planning

### PR DESCRIPTION
**Pull request by joshaber**
Our release planning as discussed in our mini-summit.

@desktop/core Does this sound like what we decided on? Did I miss anything? Is anything unclear?

/cc https://github.com/desktop/desktop/issues/2088

### CODE REVIEW

**Comment by j-f1** [comment]
> Save face

What’s that? (v1.3)

File: docs/process/release-planning.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+## Marketing Releases
+
+The following are our marketing releases as planned to date. They contain the broad features we're considering. The further a release is, the less defined it is. Releases are subject to change at any time.
+
+### 1.0
+- See the [1.0 milestone](https://github.com/desktop/desktop/milestone/7)
+
+### 1.1
+
+- Relationship to _X_
```
Comment (shiftkey) [inline docs/process/release-planning.md]: Should we expand a bit on these bullet points now to get the stuff in our heads down now, or shall we save this when it comes time to break these out into issues?

**Comment by joshaber** [comment]
It's ill-defined because it's a ways out, but the general idea is that Desktop should be preventing users from making embarrassing mistakes, e.g., committing files which still contain conflict markers, committing log statements, etc.

File: docs/process/release-planning.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+We plan work at two levels of granularity: first at the marketing release level (1.1, 1.2, etc.) and then at the milestone level. Marketing releases are made up of multiple milestones. Milestones should contain roughly a month’s worth of work.
+
+We’ll release updates as needed or when a milestone is completed.
+
+Large features and user-visible changes are feature-flagged, to be unflagged in the next marketing release. Beta and dev builds will have all feature flags enabled.
+
+## Marketing Releases
+
+The following are our marketing releases as planned to date. They contain the broad features we're considering. The further a release is, the less defined it is. Releases are subject to change at any time.
```
Comment (shiftkey) [inline docs/process/release-planning.md]: The further _away_ a release is?

File: docs/process/release-planning.md (hunk 1 above)
Comment (j-f1) [inline docs/process/release-planning.md]: You could also create projects for each release, then add notes for each item.

Comment (joshaber) [inline docs/process/release-planning.md]: Yeah we certainly could. If we end up having a need for more meta-organization we'll look at projects.

Comment (joshaber) [inline docs/process/release-planning.md]: 👍 fleshed out the bullet points a bit more.

File: docs/process/roadmap.md [hunk 3]
```diff
@@ -1,13 +1,67 @@
...
+- See the [1.0 milestone](https://github.com/desktop/desktop/milestone/7)
+
+## 1.1
+
+- Relationship to _X_
+  - Show how the current branch relates to the target branch
+  - The target is the default branch or the upstream's default branch
+  - _Probably_ let the user switch the target branch
+
+- Conflict resolution lite
```
Comment (shiftkey) [inline docs/process/roadmap.md]: We should give some context on what `lite` means when we use it to describe these features

File: docs/process/roadmap.md [hunk 4]
```diff
@@ -1,13 +1,67 @@
...
 
-Currently the team are focused on the 1.0 release, which should represent the
-set of features which enable users to replace the previous Classic Mac and
-Windows apps with this new Electron app.
+The following are our marketing releases as planned to date. They contain the broad features we're considering. The further away a release is, the less defined it is. Releases are subject to change at any time.
 
-Along with that, we'll be working on feedback and issues reported from the
-public beta. But feel free to open issues and start discussions on things
-you'd like to see in the app.
+## 1.0
```
Comment (shiftkey) [inline docs/process/roadmap.md]: Maybe some words about what 1.0 represents, especially when compared to the Classic apps?

File: docs/process/roadmap.md (hunk 3 above)
Comment (joshaber) [inline docs/process/roadmap.md]: I hope fleshing them out would make it clearer. Do you have any ideas of what other words we should use?

Comment (shiftkey) [inline docs/process/roadmap.md]: Yeah, that works for me 👍

**Pull request merged**